packages/core/src/module-resolver.ts

```diff
@@ -870,6 +871,31 @@ export class Resolver {
     return request;
   }
 
+  private handleVendor<R extends ModuleRequest>(request: R): R {
+    //TODO move the extra forwardslash handling out into the vite plugin
+    const candidates = [
+      '#embroider/core/vendor',
+      '@embroider/core/vendor',
+      '/@embroider/core/vendor',
+      './@embroider/core/vendor',
+    ];
+
+    if (!candidates.includes(request.specifier)) {
+      return request;
+    }
+
+    let pkg = this.packageCache.ownerOfFile(request.fromFile);
+    if (pkg?.root !== this.options.engines[0].root) {
+      return request;
```

Comment:
so the check here is correct but the thing that you're doing is not quite. Essentially, we only want to support somebody adding the `@embroider/core/vendor` to their app code, i.e. the package that owns the file that contains the reference to `@embroider/core/vendor` is their actual top-level app.

If we just return the request here you would get some strange error from vite or webpack saying `could not resolve @embroider/core/vendor` and it wouldn't communicate anything to the end user that something was wrong. We have an opportunity here to tell the end user **exactly** what is wrong so we should take it, and we should throw an error along the lines of `we found an import of @embroider/core/vendor in ${fromFile} but that is not in your top-level ember app. We only support blah blah blah. Open an issue etc.`